File: java/src/test/EditorUserTest.java
package mx.iteso;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Editor user test class.
 */
public class EditorUserTest {
    private User user;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        user = mock((EditorUser.class));
    }    
    /**
     * Get name test.
     */
    @Test
    public void NameTest(){
        when(user.getName()).thenReturn("Editor");
    }
}

File: csharp/decorator_practice/decorator_practice/test/EditorUserTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class EditorUserTest
    {
        public Mock<User> user;
        public EditorUser eu;

        [SetUp]
        public void SetUp()
        {
            eu = new EditorUser();
            user = new Mock<User>(eu);
        }

        [TestCase]
        public void NameTest()
        {
            Assert.AreEqual("Editor", user.Object.getName());
        }
    }
}